syntax = "proto3";
package elkr;

service ElkrService {
    rpc AddSink(AddSinkRequest) returns (AddSinkResponse);
    rpc RemoveSink(RemoveSinkRequest) returns (RemoveSinkResponse);
    rpc ListSinks(ListSinksRequest) returns (ListSinksResponse);
}

enum Code {
    CODE_NULL = 0;
    OK = 200;
    CLIENT_ERROR = 400;
    SERVER_ERROR = 500;
};

message AddSinkRequest {
    string path = 1;
};

message AddSinkResponse {
    Code code = 1;
};

message RemoveSinkRequest {
    string path = 1;
};

message RemoveSinkResponse {
    Code code = 1;
};

message ListSinksRequest {
};

message ListSinksResponse {
    Code code = 1;
    repeated string sinks = 23;
};
